From ea8aab72d26006219432bb6f32c7e88b1cdf8d57 Mon Sep 17 00:00:00 2001
From: Rich Bono <rbono@redhat.com>
Date: Thu, 20 Nov 2014 21:23:41 -0500
Subject: [scsi] pm8001: Fix erratic calculation in update_flash

Message-id: <1416518623-7080-21-git-send-email-rbono@redhat.com>
Patchwork-id: 100413
O-Subject: [RHEL7.1 e-stor PATCH V3 20/22] pm8001: Fix erratic calculation in update_flash
Bugzilla: 1110943
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tomas Henzl <thenzl@redhat.com>

From: Tomas Henzl <thenzl@redhat.com>

The loopcount is calculated by using some weird magic. Use instead a boring
macro.

Signed-off-by: Tomas Henzl <thenzl@redhat.com>
Acked-by: Suresh Thiagarajan <Suresh.Thiagarajan@pmcs.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
(cherry picked from commit 5bd355ee3b0191c6846918b049b3940182bec1cb)

Signed-off-by: Rich Bono <rbono@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/pm8001/pm8001_ctl.c b/drivers/scsi/pm8001/pm8001_ctl.c
index d3a08ae..e9b0921 100644
--- a/drivers/scsi/pm8001/pm8001_ctl.c
+++ b/drivers/scsi/pm8001/pm8001_ctl.c
@@ -581,11 +581,8 @@ static int pm8001_update_flash(struct pm8001_hba_info *pm8001_ha)
   partitionSizeTmp =
    *(u32 *)((u8 *)&image_hdr->image_length + sizeRead);
   partitionSize = be32_to_cpu(partitionSizeTmp);
-  loopcount = (partitionSize + HEADER_LEN)/IOCTL_BUF_SIZE;
-  if (loopcount % IOCTL_BUF_SIZE)
-   loopcount++;
-  if (loopcount == 0)
-   loopcount++;
+  loopcount = DIV_ROUND_UP(partitionSize + HEADER_LEN,
+     IOCTL_BUF_SIZE);
   for (loopNumber = 0; loopNumber < loopcount; loopNumber++) {
    payload = (struct pm8001_ioctl_payload *)ioctlbuffer;
    payload->length = 1024*16;
-- 
1.7.1